core/src/main/java/org/infinispan/persistence/spi/TransactionalCacheWriter.java

```diff
@@ -0,0 +1,40 @@
+package org.infinispan.persistence.spi;
+
+import org.infinispan.persistence.support.BatchModification;
+
+import javax.transaction.Transaction;
+
+/**
+ * Defines the functionality of a transactional store.  This allows the implementing store to participate in the cache's 2PC
+ * protocol to ensure consistency between the underlying store and the cache.
```

Comment:
Should we have more details here about how the consistency is affected when the originator node crashes (with/without recovery)?